### PR TITLE
[WIP]Mod loading

### PR DESCRIPTION
**Pull request by wherkamp**
- [x] Mods Loading
- [x] Basic Player Movement
- [x] Basic Chunk Loading
- [ ] Basic Main Menu
- [ ] Basic World Selection Menu
- [ ] Basic World System

### CODE REVIEW

**Comment by wherkamp** [comment]
This will be put on hold until the World and Save classes/interfaces are done.

File: client/src/main/java/org/kakara/client/scenes/MainGameScene.java [hunk 1]
```diff
@@ -1,20 +1,143 @@
...
+
+        light = new PointLight(new Vector3f(0, 2, 0));
+        //lightIndication = new MeshGameItem(mesh).setScale(0.3f).setPosition(0, 2, 0);
+        add(light);
+        this.add(new PointLight().setPosition(12, 12, 0).setDiffuse(0.1f, 0, 0).setSpecular(1.0f, 0, 0));
+        this.add(new PointLight().setPosition(3, 3, 3).setDiffuse(0f, 0.3f, 0).setSpecular(0, 0, 0.7f));
+        this.add(new SpotLight(gameHandler.getCamera().getPosition(), new Vector3(0, 0, 1)));
+        // this.add(lightIndication);
+        // Allows you to see the light.
+        this.getLightHandler().getDirectionalLight().setDirection(12, 12, 0);
```
Comment (ryandw11) [inline client/src/main/java/org/kakara/client/scenes/MainGameScene.java]: This is not pointing at the ground at all. Do  0, -1, 0 for noon effect.

File: client/src/main/java/org/kakara/client/scenes/MainGameScene.java [hunk 2]
```diff
@@ -1,20 +1,143 @@
...
+            player.movePosition(0, 0, 1);
+        }
+        if (ki.isKeyPressed(GLFW_KEY_A)) {
+            player.movePosition(-1, 0, 0);
+        }
+        if (ki.isKeyPressed(GLFW_KEY_D)) {
+            player.movePosition(1, 0, 0);
+        }
+        if (ki.isKeyPressed(GLFW_KEY_SPACE)) {
+            player.movePosition(0, 1, 0);
```
Comment (ryandw11) [inline client/src/main/java/org/kakara/client/scenes/MainGameScene.java]: Not the best way to handle jumping, but it does work.

File: client/src/main/java/org/kakara/client/scenes/MainMenuScene.java [hunk 3]
```diff
@@ -10,12 +12,17 @@
...
 import org.kakara.engine.ui.text.Font;
 import org.kakara.engine.ui.text.TextAlign;
+import org.kakara.engine.utils.Time;
 
 import java.net.MalformedURLException;
 
+import static org.lwjgl.glfw.GLFW.*;
+import static org.lwjgl.glfw.GLFW.GLFW_KEY_LEFT_SHIFT;
+
 public class MainMenuScene extends AbstractGameScene {
```
Comment (ryandw11) [inline client/src/main/java/org/kakara/client/scenes/MainMenuScene.java]: Change to AbstractMenuScreen

File: engine/src/main/java/org/kakara/engine/item/ItemHandler.java [hunk 4]
```diff
@@ -43,6 +43,10 @@ public GameItem getItemWithId(UUID id) {
         return null;
     }
 
+    public List<GameItem> getItems() {
```
Comment (ryandw11) [inline engine/src/main/java/org/kakara/engine/item/ItemHandler.java]: Why are you adding this back? The items list is empty!

File: engine/src/main/java/org/kakara/engine/item/MeshGameItem.java [hunk 5]
```diff
@@ -245,5 +249,39 @@ public GameItem clone(boolean exact) {
         return clone;
     }
 
+    public void movePosition(float offsetX, float offsetY, float offsetZ) {
```
Comment (ryandw11) [inline engine/src/main/java/org/kakara/engine/item/MeshGameItem.java]: What in the world? Why are you working with sin and cos? Shouldn't you just add the offset to the positon?

File: engine/src/main/java/org/kakara/engine/item/Texture.java [hunk 6]
```diff
@@ -79,7 +79,10 @@ public Texture(Resource resource, Scene currentScene) {
                 decodedImage = stbi_load_from_memory(resource.getByteBuffer(), w, h, avChannels, 4);
             }else{
                 decodedImage = stbi_load(resource.getPath(), w, h, avChannels, 4);
-
+            }
+            if(decodedImage==null) {
+                System.out.println("Failure");
```
Comment (ryandw11) [inline engine/src/main/java/org/kakara/engine/item/Texture.java]: Change this to a log message that specifies where this is.

File: client/src/main/java/org/kakara/client/scenes/MainGameScene.java (hunk 2 above)
Comment (wherkamp) [reply client/src/main/java/org/kakara/client/scenes/MainGameScene.java]: I know.

File: engine/src/main/java/org/kakara/engine/item/ItemHandler.java (hunk 4 above)
Comment (wherkamp) [reply engine/src/main/java/org/kakara/engine/item/ItemHandler.java]: I will remove that.

File: engine/src/main/java/org/kakara/engine/item/Texture.java (hunk 6 above)
Comment (wherkamp) [reply engine/src/main/java/org/kakara/engine/item/Texture.java]: IDK, I think Failure gets the point across. 

I will change it.

Comment (ryandw11) [reply engine/src/main/java/org/kakara/engine/item/Texture.java]: What failed though? We should be more specific.

**Comment by wherkamp** [comment]
Faster Mod Loading.

**Pull request closed without merging**